src/main/java/survey/controller/WebController.java

```diff
@@ -62,13 +62,20 @@ public Survey createNewSurvey(@RequestBody Survey survey){
         return survey;
     }
 
+    @PostMapping("/admin/{surveyId}/close")
+    public Survey closeSurvey(@PathVariable(value="surveyId") int surveyId){
+        Survey survey = repository.findById(surveyId);
+        survey.close();
+        repository.save(survey);
+        return survey;
+    }
+
     @PostMapping("/admin/{surveyId}/mcq")
     public MultipleChoiceQuestion addMcqQuestionToSurvey(@PathVariable(value="surveyId") int surveyId, @RequestBody MultipleChoiceQuestion mcq){
         Survey survey = repository.findById(surveyId);
         mcqRepository.save(mcq);
         survey.addQuestion(mcq);
         repository.save(survey);
-        repository.save(survey);
```

Comment:
DId we past the test cases on maven test locally?

Reply:
yes they passed